R/biokinetics.R

```diff
@@ -253,13 +247,17 @@ biokinetics <- R6::R6Class(
         }
         private$data <- data
       }
-      unknown_vars <- private$all_formula_vars[which(!(private$all_formula_vars %in% names(private$data)))]
-      if (length(unknown_vars) > 0) {
-        stop(paste("All variables in 'covariate_formula' must correspond to data columns. Found unknown variables:",
-                   paste(unknown_vars, collapse = ", ")))
-      }
+      validate_required_cols(private$data)
+      validate_formula_vars(private$all_formula_vars, private$data)
       logger::log_info("Preparing data for stan")
-      private$data <- convert_log_scale(private$data, "titre")
+      private$data <- convert_log_scale(private$data, "value")
+      private$data[, `:=`(titre_type_num = as.numeric(as.factor(titre_type)),
+                          obs_id = seq_len(.N))]
+      if (time_type == "relative") {
+        private$data[, t_since_last_exp := as.integer(date - last_exp_date, units = "days")]
+      } else {
+        private$data[, t_since_min_date := as.integer(date - min(date), units = "days")]
```

Comment:
creating cols `titre_type_num`, `obs_id` and `t_since_x` rather than requiring user to pass them in - basically moving more of the data processing logic into this package, so that the required inputs are as simple as possible